sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnAdapters.java

```diff
@@ -62,6 +62,15 @@ private DoFnAdapters() {}
   @SuppressWarnings({"unchecked", "rawtypes"})
   public static <InputT, OutputT> OldDoFn<InputT, OutputT> toOldDoFn(DoFn<InputT, OutputT> fn) {
     DoFnSignature signature = DoFnSignatures.INSTANCE.getOrParseSignature((Class) fn.getClass());
+
+    if (!signature.stateDeclarations().isEmpty()) {
```

Comment:
Done. I was just delaying introduction of the new path of `ParDo.of` calling `DoFnSignatures`. Later, when `ParDo` supports the feature, the check will need to move to runners to ensure they can support the features required by the `ParDo`. That might involve a different factoring of the utility code.